crates/wasm-mutate/src/info.rs

```diff
@@ -93,7 +93,7 @@ impl<'a> ModuleInfo<'a> {
 
                     // Save function types
                     for ty in reader {
-                        let typeinfo = TypeInfo::try_from(ty?).unwrap();
+                        let typeinfo = TypeInfo::try_from(ty?.structural_type).unwrap();
```

Comment:
I think it would be best to update `wasm-mutate` a bit differently in this commit. Currently `wasm-mutate` doesn't have support for GC, and that's ok, but if given a binary with gc with these changes it runs a risk of "accepting" the binary and making it invalid. For example this is ignoring all fields other than `structural_type`. 

I think it would be better to keep wasm-mutate as-is where it explicitly rejects gc-using binaries. For wasm-mutate that would look like handling the full `SubType` structure but returning errors if any fields are set that look like gc.

Reply:
Fixed.